src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -677,17 +677,26 @@ protected void SelectVersion()
                     // Select the installed version by default.
                     // Otherwise, select the first version in the version list.
                     IEnumerable<DisplayVersion> possibleVersions = _versions.Where(v => v != null);
-                    SelectedVersion =
-                        possibleVersions.FirstOrDefault(v => v.Version.Equals(_searchResultPackage?.AllowedVersions?.OriginalString))
-                        ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
+                    if (_filter.Equals(ItemFilter.UpdatesAvailable) || _filter.Equals(ItemFilter.All))
+                    {
+                        // If the latest version is not the installed version, we need to
```

Comment:
"Need to"...... what ?
The following code seems opposite of the comment. It looks like it wants to test that we *do* have the latest version.

Reply:
Probably this comment was for another thing in the past, I'll update the above comment to explain this if